File: lever/src/main/java/com/pin/lever/MemoryPolicy.java
package com.pin.lever;

public enum MemoryPolicy {

    /** Skips memory cache lookup when processing a request. */
    NO_CACHE(1),
    /**
     * Skips storing the final result into memory cache. Useful for one-off requests
     * to avoid evicting other bitmaps from the cache.
     */
    NO_STORE(1 << 1);

    static boolean shouldReadFromMemoryCache(int memoryPolicy) {
        return (memoryPolicy & MemoryPolicy.NO_CACHE.index) == 0;
    }

    static boolean shouldWriteToMemoryCache(int memoryPolicy) {
        return (memoryPolicy & MemoryPolicy.NO_STORE.index) == 0;
    }

    final int index;

    MemoryPolicy(int index) {
        this.index = index;
    }
}

File: lever/src/main/java/com/pin/lever/MemoryPolicy.kt
package com.pin.lever

enum class MemoryPolicy(val index: Int) {
    /** Skips memory cache lookup when processing a request.  */
    NO_CACHE(1),
    /**
     * Skips storing the final result into memory cache. Useful for one-off requests
     * to avoid evicting other bitmaps from the cache.
     */
    NO_STORE(1 shl 1);

    companion object {
        @JvmStatic
        fun shouldReadFromMemoryCache(memoryPolicy: Int): Boolean {
            return memoryPolicy and NO_CACHE.index == 0
        }

        @JvmStatic
        fun shouldWriteToMemoryCache(memoryPolicy: Int): Boolean {
            return memoryPolicy and NO_STORE.index == 0
        }
    }

}